src/cider/nrepl.clj

```diff
@@ -19,15 +20,28 @@
    [nrepl.server :as nrepl-server]
    [orchard.java]))
 
-;; Perform the dynamic `require` asap, and also not within a separate thread:
+;; Perform the underlying dynamic `require`s asap, and also not within a separate thread
+;; (note the `future` used in `#'initializer`),
+;; since `require` is not thread-safe:
 @@orchard.java/parser-next-available?
+@analyzer/spec-abbrev
 
 ;; Warm up this cache, drastically improving the completion and info UX performance for first hits.
-;; (This was our behavior for many years, then had to be disabled for test suite reasons in Orchard 0.15.0 to 0.17.0 / cider-nrepl 0.38.0 to 0.41.0, and now it's restored again)
+;; (This was our behavior for many years,
+;; then had to be disabled for test suite reasons in Orchard 0.15.0 to 0.17.0 / cider-nrepl 0.38.0 to 0.41.0, and now it's restored again)
 ;; Note that this can only be done cider-nrepl side, unlike before when it was done in Orchard itself.
 (def initializer
   (future
+    ;; 1.- Warmup the overall cache for core Java and Clojure stuff
     @orchard.java/cache-initializer
+    ;; 2.- Also cache members involved in a typical exception (which includes stuff from nrepl, clojure.repl, etc)
+    (try
```

Comment:
I still don't get what happens in Haystack that takes so much time, though, so we'd need this warm up code.

Reply:
(I'm also guessing we can just load explicitly whatever's not getting loaded before the first usage, instead of having to trigger an actual exception. )

Reply:
> I still don't get what happens in Haystack that takes so much time, though, so we'd need this warm up code.

It's its N `orchard/info*` paths. When Enrich is available, we are able to call the (JDK-official) Java source parser which happens to be slow - about 1s per class.

> (I'm also guessing we can just load explicitly whatever's not getting loaded before the first usage, instead of having to trigger an actual exception. )

I'd be afraid of that fixed list becoming outdated at some point, which would mean we'd have a performance regression (for first hits, that is)

Reply:
OK, fair enough. 

> Java source parser which happens to be slow - about 1s per class.

That seems way too much to me. Perhaps it can be optimized down the road.

Reply:
I agree. Last time I had a look it was essentially an issue in this official JDK offering.

I might make some noise in some mailing list, but it would feel pretty futile.